ziggurat-core-geoip/src/geoip.rs

```diff
@@ -0,0 +1,130 @@
+//use ipinfo::{IpInfo, IpInfoConfig};   // TODO(asmie): implement IpInfo provider
+use ip2location::{Record, DB};
+use ipgeolocate::{Locator, Service};
+
+/// List of supported geoip providers.
+pub enum Provider {
+    IpApiCom,
+    IpApiCo,
+    IpInfo,
+    Ip2Location,
+}
+
+/// Geoip provider service configuration.
+pub struct GeoIPService {
+    /// Geoip provider. Look at `Provider` enum for supported providers.
+    pub provider: Provider,
+    /// API key (paid or free) for online providers. For offline providers put here path to database file.
+    pub api_key: String,
+}
+
+/// IP information
+pub struct IPGeoInfo {
+    /// IP address
+    pub ip: String,
+    /// Country name (long name)
+    pub country: Option<String>,
+    /// City name
+    pub city: Option<String>,
+    /// IP address latitude
+    pub latitude: Option<f64>,
+    /// IP address longitude
+    pub longitude: Option<f64>,
+    /// Timezone of the IP
```

Comment:
we could put these into a (e.g.) `Coordinates`/`Location` struct, so when we unwrap the location, we get both  values

I assume this is always true:

```rust
assert_eq(latitude.is_some(), latitude.is_some());
assert_eq(latitude.is_none(), latitude.is_none())
```